### PR TITLE
[WIP] LKE-3844 feat(ogma-singleton): add a reset option for LKOgma to reuse an instance

### PR DESCRIPTION
**Pull request by EricMallard**

### CODE REVIEW

File: src/ogma/index.ts [hunk 1]
```diff
@@ -39,19 +39,26 @@ interface AddItemOptions {
...
+  public edgeTypeWatcher!: NonObjectPropertyWatcher<LkNodeData, LkEdgeData>;
+  public store!: OgmaStore;
+  private _configuration: IOgmaConfig;
 
   constructor(configuration: IOgmaConfig) {
     // set Ogma global configuration
     super(configuration);
+    this._configuration = configuration;
+    Object.setPrototypeOf(this, new.target.prototype);
+    this.initOgmaLinkuriusParser();
```
Comment (maxime-allex) [inline src/ogma/index.ts:58]: typo: `linkurius`

File: src/ogma/index.ts [hunk 2]
```diff
@@ -39,19 +39,26 @@ interface AddItemOptions {
 }
 
 export class LKOgma extends Ogma<LkNodeData, LkEdgeData> {
-  private _reactive: RxViz;
+  private _reactive?: RxViz;
```
Comment (maxime-allex) [inline src/ogma/index.ts:42]: Why this is optional?

File: src/ogma/index.ts [hunk 3]
```diff
@@ -39,19 +39,26 @@ interface AddItemOptions {
...
-  public nodeCategoriesWatcher: NonObjectPropertyWatcher<LkNodeData, LkEdgeData>;
+  public nodeCategoriesWatcher!: NonObjectPropertyWatcher<LkNodeData, LkEdgeData>;
   // Trigger an event with edge type changes
-  public edgeTypeWatcher: NonObjectPropertyWatcher<LkNodeData, LkEdgeData>;
-  public store: OgmaStore;
+  public edgeTypeWatcher!: NonObjectPropertyWatcher<LkNodeData, LkEdgeData>;
+  public store!: OgmaStore;
+  private _configuration: IOgmaConfig;
 
   constructor(configuration: IOgmaConfig) {
```
Comment (maxime-allex) [inline src/ogma/index.ts:53]: You can set the configuration directly in the constructor: 
```suggestion
  constructor(private _configuration: IOgmaConfig) {
```

File: src/ogma/index.ts [hunk 4]
```diff
@@ -39,19 +39,26 @@ interface AddItemOptions {
...
     // set Ogma global configuration
     super(configuration);
+    this._configuration = configuration;
+    Object.setPrototypeOf(this, new.target.prototype);
+    this.initOgmaLinkuriusParser();
+  }
+
+  private initOgmaLinkuriusParser(): void {
     this.nodeCategoriesWatcher = this.schema.watchNodeNonObjectProperty({
```
Comment (maxime-allex) [inline src/ogma/index.ts:62]: There was an old old bug in Ogma v2 where the watchers where not correctly cleared. You should check with the Ogma team how to correctly clear the schema and the watchers.

File: src/ogma/index.ts [hunk 5]
```diff
@@ -305,10 +311,21 @@ export class LKOgma extends Ogma<LkNodeData, LkEdgeData> {
...
     this.destroy();
     if (this.store) {
       this.store.clear();
     }
   }
+
+  /**
+   * Reset the Ogma instance so that it can be used fresh in the next visulization
+   */
+  public resetOgmaLKE(): void {
```
Comment (maxime-allex) [inline src/ogma/index.ts:324]: why not calling this method: `clearOgmaState`? That would be more representative of what the method is actually doing.

**Review by maxime-allex — CHANGES_REQUESTED** [review]

File: src/ogma/index.ts (hunk 4 above)
Comment (EricMallard) [reply src/ogma/index.ts:62]: https://github.com/Linkurious/ogma/issues/2827
they are on it. Ogma will make sure that everything is correctly deleted on reset. But we checked the code with @Leo-Nicolle and nothing very annoying should happen!
It is used to update captions so I will test it anyway, to be sure everything is alright!

**Review by maxime-allex — APPROVED** [review]

**Review by maxime-allex — APPROVED** [review]

**Review by maxime-allex — APPROVED** [review]

**Pull request merged**